sdks/ios/app/main.m

```diff
@@ -54,66 +55,88 @@ - (void)applicationWillTerminate:(UIApplication *)application {
 
 @implementation ViewController
 
-UIView *mainView;
-UILabel *lblResults;
+UILabel *lblTestResults, *lblSummary;
 int passed = 0, skipped = 0, failed = 0;
+char* summaryMessage = nil;
 
 - (void)viewDidLoad {
-    [super viewDidLoad];
+	[super viewDidLoad];
+	self.view.backgroundColor = [UIColor whiteColor];
+
+	//
+	// construct UI
+	//
+	UIStackView *mainStackView = [[UIStackView alloc] init];
+	mainStackView.translatesAutoresizingMaskIntoConstraints = NO;
+	mainStackView.axis = UILayoutConstraintAxisVertical;
+	mainStackView.alignment = UIStackViewAlignmentFill;
+	mainStackView.distribution = UIStackViewDistributionFill;
+	mainStackView.spacing = 8;
+
+	struct utsname systemInfo;
+	uname(&systemInfo);
+	UILabel *lblHeader = [[UILabel alloc] init];
+	lblHeader.textColor = [UIColor blueColor];
+	lblHeader.font = [UIFont boldSystemFontOfSize: 18];
+	lblHeader.text = [NSString stringWithFormat:@"Mono iOS SDK on %s", systemInfo.machine];
 
-    mainView = [[UIView alloc] initWithFrame:self.view.frame];
-    mainView.backgroundColor = [UIColor whiteColor];
-	[self.view addSubview:mainView];
+	UILabel *lblInfo = [[UILabel alloc] init];
+	lblInfo.font = [UIFont boldSystemFontOfSize: 15];
+	lblInfo.text = @"Test results:";
 
-	UILabel *lblHeader = [[UILabel alloc] init];
-    lblHeader.frame = CGRectMake(100, 100, 200, 200);
-	lblHeader.backgroundColor = [UIColor clearColor];
-	lblHeader.textColor = [UIColor redColor];
-	lblHeader.font = [UIFont boldSystemFontOfSize: 20];
-	lblHeader.numberOfLines = 2;
-	lblHeader.text = @"Mono iOS SDK\nRunning tests...";
-	[mainView addSubview:lblHeader];
-
-	lblResults = [[UILabel alloc] init];
-    lblResults.frame = CGRectMake(100, 200, 200, 200);
-	lblResults.backgroundColor = [UIColor clearColor];
-	lblResults.textColor = [UIColor redColor];
-	lblResults.font = [UIFont boldSystemFontOfSize: 20];
-	lblResults.numberOfLines = 3;
-	lblResults.text = @"Passed: 0\nSkipped: 0\nFailed: 0";
-	[mainView addSubview:lblResults];
+	lblTestResults = [[UILabel alloc] init];
+	lblTestResults.text = @"";
+	lblTestResults.numberOfLines = 3;
+
+	lblSummary = [[UILabel alloc] init];
+	lblSummary.accessibilityIdentifier = @"SummaryLabel";
+	lblSummary.text = @"Waiting for test to start.";
+
+	[mainStackView addArrangedSubview:lblHeader];
+	[mainStackView addArrangedSubview:lblInfo];
+	[mainStackView addArrangedSubview:lblTestResults];
+	[mainStackView addArrangedSubview:lblSummary];
```

Comment:
Don't test just quit right after all tests are finished so the summary will never be visible? 🙂

Reply:
Not with the changes in https://github.com/akoeplinger/mono/blob/752bfa359ad4a8f131b18382d68c977f39b6d27f/sdks/ios/test-runner/runner.cs#L291-L293 😄